### ISSUE TITLE
with fec,the max playload to 1452bytes

### ISSUE THREAD

**Issue by chllcy** [Haivision/srt#1322]
Hi,I test with fec,but encounter problem with playload size。send with 1456 bytes。1500MTU。
the log below：
Due to filter-required extra 4 bytes, SRTO_PAYLOADSIZE fixed to 1452 bytes

![image](https://user-images.githubusercontent.com/12612758/83758670-feecca80-a6a4-11ea-882f-2033bbb6e633.png)

is it need to change max payload size to 1452 if with fec feature at handshake？

**Comment by ethouris** [comment]
Did you set the payload size using `SRTO_PAYLOADSIZE`?

Might matter the order in which you set options. When you set payloadsize first and packetfilter second, the packetfilter setting should cut the payloadsize. In the opposite order, it should disallow you to set payloadsize that big.

Note also that there's #1130 PR to remove this option completely. The actual maximum payload size is then managed internally and allowed as maximum in the beginning.

**Comment by chllcy** [comment]
with send 1456 bytes payload, fec rebuild will report error.
FEC: DECLIPPED length '1456' exceeds payload size. NOT REBUILDING.

**Comment by ethouris** [comment]
Yes, definitely this is an internal error that shouldn't happen.

How did you configure the socket then? For me it looks as if the payload size was somehow allowed to be set to 1456, which shouldn't be allowed when you also configured FEC. Some check looks like have been bypassed.

**Comment by mbakholdina** [comment]
@ethouris Please do appropriate tests and let me know whether PR #1130 fixes the issue or any additional work is required.

**Comment by ethouris** [comment]
@chllcy Could you please describe the exact way how you made the test? It looks for me as if you have forcefully set the payloadsize option to 1456 on the sender, and then you tried to send 1456 bytes of data per packet. This should be rejected as error, and looks like it somehow didn't. It would be nice if you could repeat this test with using the version from #1130 PR at least on the sender side and confirm or deny as to whether this fixes the problem.

**Comment by chllcy** [comment]
yes I hard code for 1456 on the sender。is it necessary to fix this problem？
I will try the version from #1130 PR。

**Comment by ethouris** [comment]
What *should* happen is that when you configure FEC on the connection and try to send 1456 bytes through that connection, your sending function should report error and reject sending. The version from that PR simply disables the `SRTO_PAYLOADSIZE` option (it's treated as deprecated and ignored), but still it allows you to send the maximum possible number of bytes through one sending call. Of course, the maximum is 1456 in a normal situation, but it's decreased by 4 when you configure FEC. Whether the control of the maximum single sending size is in force or not, it depends this time on `SRTO_TRANSTYPE` option.

**Comment by kgbook** [comment]
The same issue.

I set `SRT_LIVE_MAX_PLSIZE` to `SRTO_PAYLOADSIZE` options both on sender and receiver side, and enable FEC with `SRTO_PACKETFILTER` option.

The receiver receive no video frame, and complaint that "payload size: 1456 exceeds maximum allowed 1452".

log :
```shell
E  [SrtLogCallback:42]:[checkTransArgs#119]14:46:11.044730/VideoEncoder*E:SRT.cc: LiveCC: payload size: 1456 exceeds maximum allowed 1452
E  [input:89]:srt_sendmsg failed, reason: Operation not supported: Incorrect use of Message API (sendmsg/recvmsg)
E  [SrtLogCallback:42]:[checkTransArgs#119]14:46:11.045254/VideoEncoder*E:SRT.cc: LiveCC: payload size: 1456 exceeds maximum allowed 1452
E  [input:89]:srt_sendmsg failed, reason: Operation not supported: Incorrect use of Message API (sendmsg/recvmsg)
```

**Comment by ethouris** [comment]
Is the same thing happening if you don't change the `SRTO_PAYLOADSIZE` option?

**Comment by kgbook** [comment]
> Is the same thing happening if you don't change the `SRTO_PAYLOADSIZE` option?

no problem if not change `SRTO_PAYLOADSIZE` option. 

I fixed the issues, no more than 1456-4 bytes payload. it works.
